lib/jnpr/junos/utils/start_shell.py

```diff
@@ -112,10 +124,11 @@ def run(self, command, this=_SHELL_PROMPT):
         # run the command and capture the output
         self.send(command)
         got = ''.join(self.wait_for(this))
-
-        # use $? to get the exit code of the command
-        self.send('echo $?')
-        rc = ''.join(self.wait_for(this))
+        rc = ''
+        if re.search(r'(%|>|#)\s?$', got) is not None:
```

Comment:
We should only send 'echo $?' if we are at a shell prompt, not a Junos prompt. We shouldn't hardcode the prompt.

Suggest this line should be:
        if re.search(r'{0}\s?$'.format(SHELL_PROMPT), got) is not None: